pkg/autodiscovery/listeners/kubelet.go

```diff
@@ -243,15 +244,24 @@ func (l *KubeletListener) createService(entity string, pod *kubelet.Pod, firstRu
 			svc.metricsExcluded = l.filters.IsExcluded(containers.MetricsFilter, container.Name, containerImage, pod.Metadata.Namespace)
 			svc.logsExcluded = l.filters.IsExcluded(containers.LogsFilter, container.Name, containerImage, pod.Metadata.Namespace)
 
+			// Cache the container name to get the corresponding ports after breaking the for-loop
 			containerName = container.Name
 
+			// Check for custom AD identifiers
+			adIdentifier := container.Name
+			if customADIdentifier, customIDFound := common.GetCustomCheckID(pod.Metadata.Annotations, container.Name); customIDFound {
```

Comment:
nit: for these two lines could use `containerName` vs. `container.Name` ?

Reply:
sure we can, but what is rationale behind? sorry if I'm missing something obvious 😅

Reply:
hm, no strong reason, perhaps just a habit i developed from python where we'd use a var to store a commonly referenced object value

Reply:
done 👍